docs/references/architecture/adr-115-predictable-block-times.md

```diff
@@ -0,0 +1,113 @@
+# ADR 115: Predictable Block Times
+
+## Changelog
+
+ - April 30 2024: Created by @melekes
+
+## Status
+
+**Proposed**
+
+## Context
+
+Let's say you're an ABCI application developer and you want to have
+predictable block times: 1 block each 6s. How would you do that?
+
+You found out that proposing a block and voting on it takes roughly 2s in your
+network. You change `timeout_commit` (_how long a node waits for additional
+precommits after committing a block, before starting on the next height_) from
+1s to 4s in the default config and ship your app with that modified config. Do
+you have predictable block times now?
+
+No - you don't. The block time will be around 6s, but it won't be exactly 6s. A
+validator could set `timeout_commit`  to 0s because it's a local parameter,
+which is not enforced by Comet. That means whenever this validator proposes a
+block, the overall block time will be 2s (not 6s!).
+
+There are other reasons why the block time might not be predictable:
+
+1. CometBFT can go into multiple rounds of consensus, which will increase the
+   block time.
+2. Network latency.
+3. Clock drifts.
+
+By now, many teams (Celestia, Berachain, Osmosis, etc.) expressed their
+interest in either a) having predictable block times OR b) having the ability
+to alter the next block time.
+
+In case of Osmosis, committing a big block might take longer than usual. They
+could increase the block time for big blocks to give the state machine some
+extra time to finish execution if that was possible.
+
+In case of Berachain, they want to have a predictable block time to match
+Ethereum's slots, which are equal to 12s.
+
+## Decision
+
+Remove `timeout_commit` in favor of a new field in `FinalizeBlockResponse`:
+`next_block_delay`. This field's semantics stays essentially the same: delay
+between this block and the time when the next block is proposed.
+
+If Comet goes into multiple rounds of consensus, the ABCI application can react
```

Comment:
Not sure if the application is aware of the commit round. In any case, different nodes may commit at different rounds.

Reply:
> different nodes may commit at different rounds.

could you please explain?

Reply:
decided_last_commit https://docs.cometbft.com/v0.38/spec/abci/abci++_methods#finalizeblock ?

Reply:
A node can commit at round 0, other might decide the same block on block 1. This happens if part of the nodes don't see the precommits of round 0 timely and proceed to round 1.

The canonical commit, in fact, is the one observed by the proposer (for the previous block) and added to the proposed block.

Reply:
> Not sure if the application is aware of the commit round.

does the ABCI dev need it? They can simply use real time, no?

Reply:
Yes, it should use real time, in particular because the duration of rounds can be pretty variable.

Reply:
But the problem of real time is, of course, that different nodes will process a block at (slightly) different times.

Reply:
I have read up to here, so what I say here might be redundant with text below.
My impression is that most applications, in order to leverage this feature:
* will need to use real --wallclock-- time (having Berachain's use case in mind, where you want to conform to regular wall-clock deadlines)
* will mandate it's nodes to have synchronized clocks (NTP, or other). This is not a big deal since we also require this for PBTS